esx_service/cli/vmdkops_admin.py

```diff
@@ -1006,15 +1006,46 @@ def tenant_access_add(args):
         error_info = "Tenant {0} does not exist".format(args.name)
         return operation_fail(error_info)
 
+    privileges = [d for d in tenant.privileges if d[auth_data_const.COL_DATASTORE] == args.datastore]
+    
+    if privileges:
+        error_info = "Privilege already exists for ({0}, {1})".format(args.name, args.datastore)
+        return operation_fail(error_info)
```

Comment:
if the requested privilege is already set, it's better to succeed as no harm is done and what is request is exactly what the end result is